### PR TITLE
seek for seq (and rseq)

### PR DESCRIPTION
**Pull request by FiV0**
This is a first cut of the `seek` op for seqs of PersistentSortedSets. I haven't done anything for CLJS yet, wanted to get some feedback first before spending more time on it. 

If you want me to the PR against your `storage` branch, let me know.

### CODE REVIEW

File: src-java/me/tonsky/persistent_sorted_set/Seq.java [hunk 1]
```diff
@@ -156,4 +156,74 @@ public Seq rseq() {
...
+          node = seq.child();
+        } else { // Leaf
+          seq = new Seq(null, this._set, seq, node, idx, this._keyTo, cmp, true, _version);
+          return seq.over() ? null : seq;
+        }
+      }
+
+    } else {
+
+      while (cmp.compare(to, minKey(node)) < 0){
```
Comment (FiV0) [inline src-java/me/tonsky/persistent_sorted_set/Seq.java]: This is a bit `awkward` here. There is currently no way to get the minimum key of any of it's descendants of a `Node` in constant time. This should probably be somehow made possible o/w `seek` for rseq will have a lot of unnecessary overhead.

**Comment by tonsky** [comment]
Looks good to me. But I suggest you rebase it on `storage` branch or wait for it to be merged (this week I hope).

Re: rseek, the only problem is that we go one level too high compared to what we would’ve done if we had fast min-key, no? If we replace `cmp.compare(to, minKey(node)) < 0` with `cmp.compare(to, keys[0]) < 0`. Or wouldn’t that work?

**Comment by FiV0** [comment]
Re:Re:rseek Right. In that case we can't shortcircuit, but I guess that is not too bad.

**Comment by FiV0** [comment]
@tonsky I rebased and added the cljs part. As there are no parent pointers in the cljs implementation I essentially start from the root every time I seek. This is of course not as efficient as walking the tree, but I hope good enough for cljs.

**Comment by tonsky** [comment]
Merged and released 0.2.1. Thank you!

**Pull request merged**